src/shared/operator_cloud_client/intent_events_sender.go

```diff
@@ -0,0 +1,232 @@
+
+		// Report events and statuses before starting the ticker
+		ies.reportIntentEvents(ctx)
+		ies.ReportIntentStatuses(ctx)
+
+		eventReportTicker := time.NewTicker(time.Second * time.Duration(viper.GetInt(otterizecloudclient.IntentEventsReportIntervalKey)))
+		statusReportTicker := time.NewTicker(time.Second * time.Duration(viper.GetInt(otterizecloudclient.IntentStatusReportIntervalKey)))
+
+		for {
+			select {
+			case <-ctx.Done():
+				return
+			case <-eventReportTicker.C:
+				ies.reportIntentEvents(ctx)
+			case <-statusReportTicker.C:
+				ies.ReportIntentStatuses(ctx)
+			}
+		}
+	}()
+	return nil
+}
+
+func (ies *IntentEventsPeriodicReporter) reportIntentEvents(ctx context.Context) {
+	events := v1.EventList{}
+	gqlEvents := make([]graphqlclient.ClientIntentEventInput, 0)
+	err := ies.k8sClient.List(ctx, &events, client.MatchingFields{involvedObjectKindField: "ClientIntents"})
+	if err != nil {
+		logrus.Errorf("Failed to list events: %v", err)
+		return
+	}
+	if len(events.Items) == 0 {
+		logrus.Debugf("No events in list")
+		return
+	}
+	for _, event := range events.Items {
+		key := eventKey(event.UID)
+		if cachedGeneration, ok := ies.eventCache.Get(key); ok && cachedGeneration == eventGeneration(event.Generation) {
+			continue
+
+		}
+		intent := v2alpha1.ClientIntents{}
+		err := ies.k8sClient.Get(ctx, client.ObjectKey{Namespace: event.InvolvedObject.Namespace, Name: event.InvolvedObject.Name}, &intent)
+		if err != nil {
+			logrus.Errorf("Failed to get intent %s/%s: %v", event.InvolvedObject.Namespace, event.InvolvedObject.Name, err)
+			continue
+		}
+		si := intent.ToServiceIdentity()
+
+		gqlEvents = append(gqlEvents, graphqlclient.ClientIntentEventInput{
+			ClientName:         si.Name,
+			ClientWorkloadKind: si.Kind,
+			Namespace:          si.Namespace,
+			Name:               event.Name,
+			Labels:             convertMapToKVInput(event.Labels),
+			Annotations:        convertMapToKVInput(event.Annotations),
+			Count:              int(event.Count),
+			ClientIntentName:   event.InvolvedObject.Name,
+			FirstTimestamp:     event.FirstTimestamp.Time,
+			LastTimestamp:      event.LastTimestamp.Time,
+			ReportingComponent: event.ReportingController,
+			ReportingInstance:  event.ReportingInstance,
+			SourceComponent:    event.Source.Component,
+			Type:               event.Type,
+			Reason:             event.Reason,
+			Message:            event.Message,
+		})
+		ies.eventCache.Add(key, eventGeneration(event.Generation))
+	}
+
+	if len(gqlEvents) == 0 {
+		logrus.Debugf("No new intent events to report; events in cache: %d", ies.eventCache.Len())
+		return
+	}
+
+	timeoutCtx, cancel := context.WithTimeout(ctx, viper.GetDuration(otterizecloudclient.CloudClientTimeoutKey))
+	defer cancel()
+	eventChunks := lo.Chunk(gqlEvents, 100)
+	for _, chunk := range eventChunks {
+		ies.cloudClient.ReportIntentEvents(timeoutCtx, chunk)
+	}
+
+}
+
+func (ies *IntentEventsPeriodicReporter) ReportIntentStatuses(ctx context.Context) {
```

Comment:
some similar comments here re code organization and splitting to functions, not repeating them again but please apply here too.